portfolio/src/main/webapp/script.js

```diff
@@ -15,15 +15,44 @@
 /**
  * Adds a random greeting to the page.
  */
- function addRandomGreeting() {
-    const greetings =
-        ['Would I rather be feared or loved? Easy. Both. I want people to be afraid of how much they love me.', 'There’s a lot of beauty in ordinary things. Isn’t that kind of the point?', 'Sometimes I’ll start a sentence and I don’t even know where it’s going. I just hope I find it along the way.', 'I guess I’ve been working so hard, I forgot what it’s like to be hardly working.', 'I am Beyonce, always.'];
-  
+
+$(document).ready(function() {
+    $(window).scroll(function() {
+            if (this.scrollY > 20) {
+                $('.navbar').addClass("sticky");
+            } else {
+                $('.navbar').removeClass("sticky");
+            }
+        })
+        //for the menubar
+    $('.menu-bar').click(function() {
+            $('.navbar .links').toggleClass("active");
+            $('.menu-bar i').toggleClass("active");
+        })
+        //for the project links
+    $('.chart-wrapper').click(function() {
```

Comment:
Is this the right event? On click instead of scrolling into view or something?

Reply:
i couldnt figure out how to make it work on scroll

Reply:
have you tried `.on('scroll', ...)`?

Reply:
What you want here is a listener on scroll 

https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event

In your scroll event listener, you want to check if the element is in view, then do whatever you need to do. Same thing if you want to do something when it goes out of view.

Reply:
There are some example libraries that do this 'in view' detection, but you should be able to cobble together something using the scroll events, the viewport rect and the element rect.

Reply:
okay i'll try to look into it more